process/smartContract/process.go

```diff
@@ -566,59 +604,63 @@ func (sc *scProcessor) processSCOutputAccounts(outputAccounts []*vmcommon.Output
 			//SC with data variables
 			err := sc.accounts.SaveDataTrie(acc)
 			if err != nil {
-				return nil, err
+				return err
 			}
 		}
 
+		// change code if there is a change
 		if len(outAcc.Code) > 0 {
 			err = sc.accounts.PutCode(acc, outAcc.Code)
 			if err != nil {
-				return nil, err
+				return err
 			}
 
-			//TODO remove this when receipts are implemented
-			log.Info(fmt.Sprintf("*** Generated/called SC account: %s ***", hex.EncodeToString(outAcc.Address)))
+			fmt.Printf("Created SC addess %s \n", hex.EncodeToString(outAcc.Address))
 		}
 
-		if outAcc.Nonce == nil || outAcc.Nonce.Cmp(big.NewInt(int64(acc.GetNonce()))) < 0 {
-			return nil, process.ErrWrongNonceInVMOutput
+		// change nonce only if there is a change
+		if outAcc.Nonce > 0 {
+			if outAcc.Nonce < acc.GetNonce() {
+				return process.ErrWrongNonceInVMOutput
+			}
+
+			err = acc.SetNonceWithJournal(outAcc.Nonce)
+			if err != nil {
+				return err
+			}
 		}
 
-		err = acc.SetNonceWithJournal(outAcc.Nonce.Uint64())
-		if err != nil {
-			return nil, err
+		// if no change then continue
+		if outAcc.BalanceDelta == nil || outAcc.BalanceDelta.Cmp(zero) == 0 {
+			continue
 		}
 
-		if outAcc.Balance == nil {
-			return nil, process.ErrNilBalanceFromSC
+		// TODO: delete this when VM is resolved
+		if bytes.Equal(outAcc.Address, tx.SndAddr) {
+			outAcc.BalanceDelta = outAcc.BalanceDelta.Add(outAcc.BalanceDelta, tx.Value)
```

Comment:
why + for the sender?

Reply:
this is a quick fix for IELE VM, because it makes mistakes.